code/modules/mining/machine_redemption.dm

```diff
@@ -506,6 +512,36 @@
 							"<span class='notice'>You insert [I] into [src].</span>")
 	return TRUE
 
+/obj/machinery/mineral/ore_redemption/proc/try_refill_storage(mob/living/silicon/robot/robot)
+	. = FALSE
+	if(!istype(robot))
+		return
+	if(!istype(robot.module, /obj/item/robot_module/engineering)) // Should only happen for drones
+		return
+
+	for(var/datum/robot_storage/material/mat_store in robot.module.material_storages)
+		if(mat_store.amount == mat_store.max_amount) // Already full, no need to run a check
+			to_chat(robot, "<span class='notice'>[mat_store] could not be filled due to it already being full.</span>")
+			continue
+		var/datum/component/material_container/container_component = GetComponent(/datum/component/material_container)
+		for(var/mat_id in container_component.materials)
+			var/datum/material/stack = container_component.materials[mat_id] // Should have only `/datum/material` in the list
+			var/obj/item/stack/sheet/sheet = stack.sheet_type
+			if(ispath(mat_store.stack, sheet))
+				var/amount_to_add
+				if(stack.amount >= (mat_store.max_amount - mat_store.amount))
+					amount_to_add = mat_store.max_amount - mat_store.amount
+					mat_store.amount = mat_store.max_amount
+					stack.amount -= amount_to_add * MINERAL_MATERIAL_AMOUNT
+					to_chat(robot, "You refill [mat_store] to full.")
+				else
+					amount_to_add = round(stack.amount) // In case we have half a sheet stored
```

Comment:
The `stack.amount` is multiplied by `MINERAL_MATERIAL_AMOUNT` right?
Line 474 also has this calculation, might be good to make a proc for this so they all use the same calculations.

Reply:
Sounds good, will do in a bit